app/models/book.rb

```diff
@@ -16,4 +20,12 @@ def strip_input_fields
     self.isbn10 = isbn10.strip unless isbn10.nil?
     self.isbn13 = isbn13.strip unless isbn13.nil?
   end
+
+  def handle_status_and_borrower_correlation
+    # If the borrower is already set, do not do anything.
+    return if status != 'available' && borrower.present?
```

Comment:
If you introduce a new status at a later stage (e.g. let's say status `unavailable` for when an owner wants to temporarily make a book unavailable for any reason), then the `status != 'available'` condition will still be satisfied here, which means that the borrower will be set. 

Therefore, it is better to be explicit about the statuses that we want to save a borrower for (so `reserved` and `booked`).
e.g. you could defined the list of statuses which require a borrower, and check if the book's status is in that list
```ruby
STATUSES_REQUIRING_A_BORROWER = [status[:reserved], status[:borrowed]]

def handle_status_and_borrower_correlation
  return if STATUSES_REQUIRING_A_BORROWER.includes(status) && borrower.present?

  self.borrower = !STATUSES_REQUIRING_A_BORROWER.includes(status) ? nil : user
end
```